app/services/authentication/authenticator.rb

```diff
@@ -147,9 +147,10 @@ def update_profile_updated_at(user)
     end
 
     def account_less_than_a_week_old?(user, logged_in_identity)
-      provider_created_at = user.public_send(provider.user_created_at_field)
-      user_identity_age = provider_created_at
-      user_identity_age ||= extract_created_at_from_payload(logged_in_identity)
+      identity = user.identities.find_by(provider: provider.name)
+      provider_created_at = identity&.auth_data_dump&.dig("extra", "raw_info", "created_at")
```

Comment:
@rhymes I changed this back to something more resembling the old logic. I know I can change method calls btw, I opted for `dig` here because I find it easier to read than multiple `&.`.

Reply:
Isn't this doing the same thing that `extract_created_at_from_payload(logged_in_identity)` does ?

`logged_in_identity` and `identity` are the same object as they are extracted from the same `provider`

See also https://github.com/forem/forem/blob/f136c0ef638106c3061716505c885f692b21fde8/app/services/authentication/authenticator.rb#L28

Reply:
Ugh, it is indeed the same 🤦‍♂️ I simplified the code further, regarding the wider ramifications I guess we have to wait and see, unless @benhalpern has any insights right now.